R/add_level.R

```diff
@@ -42,21 +42,33 @@ add_top_level_internal <- function(N = NULL, ID_label = NULL,
   working_data_list[[ID_label]] <- generate_id_pad(N)
 
 
-  check_variables_named(data_arguments)
+  # check_variables_named(data_arguments)
 
   # Loop through each of the variable generating arguments
-  for (i in names(data_arguments)) {
+  for (i in seq_along(data_arguments)) {
     # Evaluate the formula in an environment consisting of:
     # 1) The current working data list
     # 2) A list that tells everyone what N means in this context.
-    working_data_list[[i]] <- expand_or_error(eval_tidy(
+    tmp <- expand_or_error(eval_tidy(
       data_arguments[[i]],
       append(working_data_list, list(N = N))
     ), N, i, data_arguments[[i]])
 
+    if(names(data_arguments)[i] != "") {
+      working_data_list[[names(data_arguments)[i]]] <- tmp
+    } else {
+      working_data_list <- c(working_data_list, tmp)
```

Comment:
Here you will need to loop over tmp explicitly and copy the items over one by one. 

Try

> c(list(a=1), list(a=2,b=3))

to see why.